### DISCUSSION TITLE
ttkbootstrap DateEntry set_date()?

### DISCUSSION THREAD

**Discussion by windir19** [israel-dryer/ttkbootstrap#498] [Q&A] [answered]
Hey there. 

I am trying to do something where I set the date in the entry field for the ttkbootstrap DateEntry. I tried using set_date() but I get this: "AttributeError: 'DateEntry' object has no attribute 'set_date'"

Looking on Stack Overflow, there was [a post that hints at the issue](https://stackoverflow.com/questions/71702681/cannot-get-value-from-dateentry-widget-in-tkinter-window-object-has-no-attrib). It seems that that ttkbootstrap DateEntry is not the same as the DateEntry for tkcalendar.DateEntry, so set_date() isn't going to work. I tried looking in the documentation and there is nothing that seems to explain or help with what I need to do.

How do I get it so I can set the date for my ttkbootstrap.DateEntry?

Any assistance is helpful. Thank you!

**Comment by rdbende** [comment]
Hi!

You should use the `startdate` keyword argument to set the date.

```py
import ttkbootstrap as ttk
import datetime

root = ttk.Window()

some_date = datetime.datetime.fromisoformat("2023-11-11")
date_chooser = ttk.DateEntry(root, startdate=some_date)
date_chooser.pack()

root.mainloop()
```

The above code should result in this window:
![grafik](https://github.com/israel-dryer/ttkbootstrap/assets/77941087/2678126a-c3dc-483f-8557-546ce93c7e15)

**Reply by windir19** [reply]
That would only be useful if I am creating the DateEntry widget right? 
What I want to do is on a button click check for a datetime somewhere else and set that date in the DateEntry field. That is why I want to use something like date_set().

**Reply by rdbende** [reply]
Indeed. If you want to set the date afterwards, you can use the `configure` method with the same keyword argument.

```py
some_other_date = datetime.datetime.fromisoformat("2023-11-12")
date_chooser.configure(startdate=some_other_date)
```

However, there seems to be a bug, because this doesn't work for me :((

**Reply by windir19** [reply]
Yeah I tried it too and while I do not get the error I had before, it doesn't change the DateEntry field for me. :(
Thank you for the input though. Hopefully one of the other developers have some input.

**Reply by Megsman** [reply]
to change the date on an existing DateEntry for example exp_date:
`exp_date.entry.insert(0, 'new_date')`

the new_date variable should be a date object.

**Reply by Ddady1** [reply]
when i try the workaround @Megsman suggested, it adds the new date to the one in the widget. So if i picked up a date let say '1/1/24' and the changed date is 1/1/25
after the insert command it shows '1/1/25/1/1/24'

**Reply by Ddady1** [reply]
Ok, i managed to workaround the issue by doing `exp_date.entry.delete(0, 'end')` before the insert.

**Comment by Megsman** [comment]
My workaround is to create a new instance of DateEntry with a startdate as option.

**Comment by israel-dryer** [comment, marked as answer]
Resolved - `DateEntry` now has a proper `set_date()` method (added after this thread), so you don't need the delete/insert dance:

```python
import datetime
date_chooser.set_date(datetime.date(2024, 1, 1))  # accepts date or datetime
```

This updates both the visible field and the widget's internal state, and works whether the widget is enabled or disabled. To read the current value back, use `get_date()`.

A few clarification for anyone landing here:

- `startdate=` is a constructor argument for the initial value - `ttk.DateEntry(root, startdate=some_date)`
- `configure(startdate=...)` does not move the displayed date - it only sets the internal fallback value. Use `set_date()` to change what's shown. (This was the source of the confusion above; in 2.0 the option is renamed `state_date` and this start_date-vs-displayed-value split is being made explicit - see #1151)
- If you're on an older release without `set_date()`, @Ddady1's workaround is the correct one.
